pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -175,18 +175,28 @@ func (p *DNSProxy) GetRules(endpointID uint16) restore.DNSRules {
 
 	restored := make(restore.DNSRules)
 	for port, entries := range p.allowed[uint64(endpointID)] {
+		count := 0
 		var ipRules restore.IPRules
 		for cs, regex := range entries {
 			var IPs map[string]struct{}
 			if !cs.IsWildcard() {
 				IPs = make(map[string]struct{})
+			Loop:
 				for _, nid := range cs.GetSelections() {
 					for _, ip := range p.LookupIPsBySecID(nid) {
 						IPs[ip] = struct{}{}
+						count++
+						if count > 1000 {
+							log.Warningf("Too many DNS rules or destinations for a port, skipping the rest")
```

Comment:
Can we include some more details here for debugging, such as endpoint ID? I guess port may not necessarily be all that interesting as it's typically 53 but we could include that too. maybe `cs`?

Reply:
Sure, a moment...

Reply:
Oh and `Warning()`, not `Warningf()`.

Reply:
Done.